pkg/controllers/disruption/controller.go

```diff
@@ -114,7 +114,9 @@ func (c *Controller) Reconcile(ctx context.Context, _ reconcile.Request) (reconc
 	// with making any scheduling decision off of our state nodes. Otherwise, we have the potential to make
 	// a scheduling decision based on a smaller subset of nodes in our cluster state than actually exist.
 	if !c.cluster.Synced(ctx) {
-		logging.FromContext(ctx).Debugf("waiting on cluster sync")
+		if time.Since(c.cluster.LastSyncTime()) > state.AbnormalClusterSyncDuration {
```

Comment:
Look at `pretty.ChangeMonitor`. It offers you a chance to leverage a cache that validates if a key is changed and you can use that as a signal to know whether you should log or not here.

Reply:
That should prevent you spamming the log line a bunch of times after you first log it once